File: me_daily/2025/12/11/cfRound1070div2/c.py
from collections import defaultdict
from typing import List
import sys
from collections import Counter
input = lambda: sys.stdin.readline().strip()
II = lambda: int(input())
LII = lambda: list(map(int, input().split()))
SI = lambda: input()
LSI = lambda: input().split()


t = II()

for _ in range(t):
    n = II()
    a = LII()
    odd = [x for x in a if x & 1]
    even = [x for x in a if x % 2 == 0]
    if len(odd) == 0:
        print(' '.join(['0'] * n))
        continue
    max_odd = max(odd)
    even.sort(reverse=True)
    pre = [0]
    for x in even:
        pre.append(pre[-1] + x)
    num_e = len(even)
    O_prime = len(odd) - 1
    ans = []
    for k in range(1, n + 1):
        L = max(0, k - 1 - num_e)
        max_ef = min(k - 1, num_e)
        if L % 2 == 0:
            ef = max_ef
        else:
            if max_ef - 1 < 0 or O_prime <= L:
                ef = -1
            else:
                ef = max_ef - 1
        if ef == -1:
            ans.append(0)
        else:
            ans.append(max_odd + pre[ef])
    print(' '.join(map(str, ans)))

File: me_daily/2025/12/11/cfRound1070div2/c.cpp
#include <bits/stdc++.h>
using namespace std;

using ll = long long;
using vll = vector<ll>;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t; cin >> t;
    while (t--) {
        int n; cin >> n;
        vll a(n);
        for (int i = 0; i < n; i++) cin >> a[i];

        vll odd, even;
        for (auto x : a) {
            if (x & 1) odd.push_back(x);
            else even.push_back(x);
        }

        if (odd.empty()) {
            for (int i = 0; i < n; i++) cout << 0 << " ";
            cout << "\n";
            continue;
        }

        sort(even.begin(), even.end(), greater<ll>());
        int m = even.size();

        vll pre(m + 1, 0);
        for (int i = 0; i < m; i++) pre[i + 1] = pre[i] + even[i];

        ll mx_odd = *max_element(odd.begin(), odd.end());
        int o_p = odd.size() - 1;

        vll ans;

        for (int k = 1; k <= n; k++) {

            int L = max(0, k - 1 - m);
            int mx_ef = min(k - 1, m);

            int ef = -1;

            if (L % 2 == 0) {
                ef = mx_ef;
            } else {
                if (mx_ef - 1 < 0 || o_p <= L)
                    ef = -1;
                else
                    ef = mx_ef - 1;
            }

            if (ef == -1) ans.push_back(0);
            else ans.push_back(mx_odd + pre[ef]);
        }

        for (auto x : ans) cout << x << " ";
        cout << "\n";
    }
}
